internal/build/phases.go

```diff
@@ -15,17 +15,32 @@ const (
 	platformDir    = "/platform"
 )
 
-func (l *Lifecycle) Detect(ctx context.Context, networkMode string, volumes []string) error {
-	detect, err := l.NewPhase(
+type RunnerCleaner interface {
+	Cleanup() error
+	Run(ctx context.Context) error
+}
+
+type PhaseManager interface {
```

Comment:
I like the idea of something that creates a `Phase` but not sure manager is the right pattern to use here. See other comments. In my opinion, this should be more like a `Factory` (or `Builder` if you want some DSL and tracking capabilities for testing).

Reply:
Agree the wording isn't quite right. We can come back to this when we settle on the scope of the interface.

Reply:
Changing to `Factory` (we have enough builders in this code base!)